$NetBSD: patch-aa,v 1.6 2015/12/29 23:34:49 dholland Exp $

Recognize x86_64 and powerpc.
Recognize Dragonfly and NetBSD.
Honor pkgsrc PTHREAD_LDFLAGS and PTHREAD_LIBS.
Use pkgsrc readline.

--- configure.orig	1999-03-15 21:03:46.000000000 -0500
+++ configure	2008-09-07 02:58:26.000000000 -0400
@@ -3250,7 +3250,7 @@ esac
 
 if test x"${JNITYPES_HOST+set}"x != x"set"x; then
     case "${host}" in
-    i?86-* | alpha-* | sparc-* | mips-* | hppa1*)
+    i?86-* | x86_64-* | alpha-* | sparc-* | mips-* | hppa1* | powerpc-*)
 	JNITYPES_HOST="${srcdir}/lib/libjni/include/jnitypes.frag";;
     *)
       { echo "configure: error: Unsupported platform.\n  Check ${srcdir}/lib/libjni/include/jnitypes.frag to see if types are correct, and\n  update configure.in." 1>&2; exit 1; }
@@ -3498,6 +3498,8 @@ case "$host" in
   mips-sgi-irix5.* | mips-sgi-irix6.*) can_use_libffi=true;;
   i*86-pc-linux*) can_use_libffi=true;;
   i*86-*-freebsd*) can_use_libffi=true;;
+  *-*-dragonfly*) can_use_libffi=true;;
+  *-*-netbsd*) can_use_libffi=true;;
   i*86-pc-solaris2*) can_use_libffi=true;;
   sparc-sun-4*) can_use_libffi=true;;
   sparc-sun-*) can_use_libffi=true;;
@@ -4043,14 +4044,14 @@ done
 fi
 
 if test "x${threads}" = xno; then
-    echo $ac_n "checking for pthread_create in -lpthread""... $ac_c" 1>&6
-echo "configure:4048: checking for pthread_create in -lpthread" >&5
+    echo $ac_n "checking for pthread_create in ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}""... $ac_c" 1>&6
+echo "configure:4048: checking for pthread_create in ${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}" >&5
 ac_lib_var=`echo pthread'_'pthread_create | sed 'y%./+-%__p_%'`
 if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lpthread  $LIBS"
+LIBS="${PTHREAD_LDFLAGS} ${PTHREAD_LIBS}  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 4056 "configure"
 #include "confdefs.h"
@@ -4084,7 +4085,7 @@ if eval "test \"`echo '$ac_cv_lib_'$ac_l
 #define $ac_tr_lib 1
 EOF
 
-  LIBS="-lpthread $LIBS"
+  LIBS="${PTHREAD_LDFLAGS} ${PTHREAD_LIBS} $LIBS"
 
 else
   echo "$ac_t""no" 1>&6
@@ -6178,7 +6179,7 @@ EOF
 
 fi
 
-subdirs="lib/libffi progs/japhard/readline"
+subdirs="lib/libffi"
 
 
 trap '' 1 2 15
@@ -6707,7 +6710,7 @@ if test "$no_recursion" != yes; then
     esac
   done
 
-  for ac_config_dir in lib/libffi progs/japhard/readline; do
+  for ac_config_dir in lib/libffi; do
 
     # Do not complain, so a configure script can configure whichever
     # parts of a large source tree are present.
